jwst/assign_wcs/util.py

```diff
@@ -366,15 +366,32 @@ def compute_fiducial(wcslist, bounding_box=None):
     spatial_footprint = footprints[spatial_axes]
     spectral_footprint = footprints[spectral_axes]
 
+    n_input = len(wcslist)
     fiducial = np.empty(len(axes_types))
+    lon = np.empty(n_input * 2)
+    lat = np.empty(n_input * 2)
     if spatial_footprint.any():
-        lon, lat = spatial_footprint
+        for i in range(n_input):
+            wcs_init = wcslist[i]
+            domain_bounds = np.hstack([wcs_init.backward_transform(
+                *w.footprint(bounding_box=bounding_box).T) for w in wcslist])
+
+            output_bounding_box = []
+            for axis in wcslist[i].output_frame.axes_order:
+                axis_min, axis_max = domain_bounds[axis].min(), domain_bounds[axis].max()
+                output_bounding_box.append((axis_min, axis_max))
+
+            sky_bounds = np.array(wcs_init.forward_transform(*output_bounding_box))
+            lon[i * 2:(i + 1) * 2] = sky_bounds[0]
+            lat[i * 2:(i + 1) * 2] = sky_bounds[1]
         lon, lat = np.deg2rad(lon), np.deg2rad(lat)
-        x_mean = np.mean(np.cos(lat) * np.cos(lon))
-        y_mean = np.mean(np.cos(lat) * np.sin(lon))
-        z_mean = np.mean(np.sin(lat))
-        lon_fiducial = np.rad2deg(np.arctan2(y_mean, x_mean)) % 360.0
-        lat_fiducial = np.rad2deg(np.arctan2(z_mean, np.sqrt(x_mean ** 2 + y_mean ** 2)))
+        x_mid = (np.max(np.cos(lat) * np.cos(lon)) +
+                 np.min(np.cos(lat) * np.cos(lon))) / 2.
+        y_mid = (np.max(np.cos(lat) * np.sin(lon)) +
+                 np.min(np.cos(lat) * np.sin(lon))) / 2.
+        z_mid = (np.max(np.sin(lat)) + np.min(np.sin(lat))) / 2.
```

Comment:
It may make sense to precompute `cos(lat)`, `cos(lon)`, ... , and use them without recomputing.